.github/workflows/release.yml

```diff
@@ -30,11 +30,15 @@ jobs:
         env:
           REPO: ${{ github.repository }}
         run: |
-          set -euo pipefail
+          set -euxo pipefail
           output=$(curl --fail-with-body -sS \
             -H "Accept: application/vnd.github+json" \
-            -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
-            "https://api.github.com/repos/$REPO/actions/workflows/postsubmit.yml/runs?per_page=1&branch=main&status=success")
+            "https://api.github.com/repos/$REPO/actions/workflows/postsubmit.yml/runs?per_page=1&branch=main&event=schedule&status=success")
+          repo_id=$(jq -r '.workflow_runs[0].head_repository.id' <<< $output)
+          if [[ "${repo_id}" != "${{ github.repository_id }}" ]] ; then
+            echo >&2 "Unexpected head repository ID: ${repo_id} - check postsubmit.yml configuration"
+            exit 1
```

Comment:
With this a malicious actor could purposely block releases for several days. Consider requesting more results from the API or iterating over it.

Reply:
I considered this but they could still block releases if we request more than one (it's hard for us to put a bound on how many malicious action runs they could trigger). Also the `event=schedule` limit should protect us from the DoS anyway.

Reply:
Oh indeed, with the `event=schedule` this should always return a valid result. So reaching that code path should never happen but it is a good additional layer of protection (in case someone changes the query in the future when a nightly release might no longer be sufficient).